railties/lib/rails/commands/server.rb

```diff
@@ -94,7 +94,8 @@ def default_options
         environment:        (ENV['RAILS_ENV'] || ENV['RACK_ENV'] || "development").dup,
         daemonize:          false,
         caching:            nil,
-        pid:                Options::DEFAULT_PID_PATH
+        pid:                Options::DEFAULT_PID_PATH,
+        restart_cmd:        'bin/rails s'
```

Comment:
@rubys I fixed it here - https://github.com/rails/rails/pull/24331/commits/233798977f5b01016e7ce40aa50c547a183bf9b8.

Now it works as expected.

```
$ rails server -b 0.0.0.0 -p 4567
=> Booting Puma
=> Rails 5.0.0.beta3 application starting in development on http://0.0.0.0:4567
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.2.0 (ruby 2.3.0-p0), codename: Spring Is A Heliocentric Viewpoint
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:4567
Use Ctrl-C to stop
* Restarting...
=> Booting Puma
=> Rails 5.0.0.beta3 application starting in development on http://0.0.0.0:4567
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.2.0 (ruby 2.3.0-p0), codename: Spring Is A Heliocentric Viewpoint
* Min threads: 5, max threads: 5
* Environment: development
* Inherited tcp://0.0.0.0:4567
Use Ctrl-C to stop
```